### PR TITLE
Account and Permissions Documentation - IFC-777

### PR DESCRIPTION
**Pull request by BeArchiTek**
- Add Permissions Topic
- Add Permissions Reference
- Update Authentication Topic
- Add Permission Guide (need to add Screenshots)

### CODE REVIEW

File: docs/docs/topics/permissions-roles.mdx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+:::
+
+### Object permissions
+
+**ObjectPermission** specifies actions, that apply to a certain kind of object. Actions like create, update, remove, and view are supported. Depending on the kind of object or branch, object permissions may be granted or refused.
+
+**Key features**:
+
+- **Supports wildcards (`*`)** to apply permissions across multiple object types.
+- **Can define different permissions per branch.**
```
Comment (BaptisteGi) [inline docs/docs/topics/permissions-roles.mdx]: Maybe we could add an example to that in the following table

Comment (BaptisteGi) [reply docs/docs/topics/permissions-roles.mdx]: Or maybe the confusion comes from the fact that we can't really set it per branch but per "branch type" default / non default

File: docs/docs/topics/permissions-roles.mdx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+:::info Example:
+
+Here are some examples of object permissions and their descriptions:
+
+| Identifier                                    | Object Type   | Action  | Decision    | Description                                                                                 |
+|-----------------------------------------------|---------------|---------|-------------|---------------------------------------------------------------------------------------------|
+| `object:*:*:create:allow_other`               | `*` (all types)| `any`   | `allow_other`| Allows creating any object, but only on non-default branches. |
+| `object:*:*:view:allow_all`                   | `*` (all types)| `view`  | `allow_all` | Allows viewing any object, anywhere, across both default and non-default branches. |
+| `object:Builtin:Tag:update:Deny`              | `BuiltinTag`   | `update`| `deny`      | Denies the ability to update any object of type BuiltinTag, across all branches. |
```
Comment (BaptisteGi) [inline docs/docs/topics/permissions-roles.mdx]:
```suggestion
| `object:Builtin:Tag:update:deny`              | `BuiltinTag`   | `update`| `deny`      | Denies the ability to update any object of type BuiltinTag, across all branches. |
```

File: docs/docs/guides/accounts-permissions.mdx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+          hfid
+        }
+      }
+    }
+    ```
+
+  </TabItem>
+</Tabs>
+
+## Managing permissions
```
Comment (BaptisteGi) [inline docs/docs/guides/accounts-permissions.mdx:131]: How can I create permission object then?

File: docs/docs/reference/permissions.mdx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Object permissions
+
+Object permissions can be applied to different types of objects and across different branches.
+
+| Identifier                                    | Object Type   | Action  | Decision    | Description                                                                                 |
+|-----------------------------------------------|---------------|---------|-------------|---------------------------------------------------------------------------------------------|
+| `object:*:*:create:allow_other`               | `*` (all types)| `any`   | `allow_other`| Allows creating any object, but only on non-default branches. |
+| `object:*:*:view:allow_all`                   | `*` (all types)| `view`  | `allow_all` | Allows viewing any object, anywhere, across both default and non-default branches. |
+| `object:Builtin:Tag:update:Deny`              | `BuiltinTag`   | `update`| `deny`      | Denies the ability to update any object of type BuiltinTag, across all branches. |
```
Comment (BaptisteGi) [inline docs/docs/reference/permissions.mdx]:
```suggestion
| `object:Builtin:Tag:update:deny`              | `BuiltinTag`   | `update`| `deny`      | Denies the ability to update any object of type BuiltinTag, across all branches. |
```

File: docs/docs/topics/auth.mdx [hunk 5]
```diff
@@ -29,14 +21,62 @@ Infrahub supports two authentication methods
...
 While using the API, the authentication token must be provided in the header:
 
 ```yaml
-X-INFRAHUB-KEY: 06438eb2-8019-4776-878c-0941b1f1d1ec
-```
+X-INFRAHUB-KEY: 06438eb2-8019-4776-878c-0941b1f1d1ec```
 
 :::
+
+## Assigning permissions
```
Comment (BaptisteGi) [inline docs/docs/topics/auth.mdx]: Should we keep the permission part in "authentication" topic? Shouldn't we create a new "authorization" or similar topic? or move it to permission/roles topic

Comment (BeArchiTek) [reply docs/docs/topics/auth.mdx]: I hesitated for this one, I think it was important to bring the gap between auth and permissions here, specially with the `default setup`, even if the permissions explanation are in the other file.

Could have a link from permission to here as well 🤔

File: docs/docs/guides/accounts-permissions.mdx (hunk 3 above)
Comment (BeArchiTek) [reply docs/docs/guides/accounts-permissions.mdx:131]: Added. Even if the mutation are not 100% the good ones. Considered the incoming PR from @gmazoyer (uniqueness_constraint, removal of name, etc) as already here to not redo it

**Review by gmazoyer — APPROVED** [review]

**Pull request merged**